experimental/rsocket-src/RSocketRequester.cpp

```diff
@@ -37,10 +37,40 @@ RSocketRequester::~RSocketRequester() {
   LOG(INFO) << "RSocketRequester => destroy";
 }
 
-std::shared_ptr<Subscriber<Payload>> RSocketRequester::requestChannel(
-    std::shared_ptr<Subscriber<Payload>> responseSink) {
-  // TODO need to runInEventBaseThread like other request methods
-  return reactiveSocket_->requestChannel(std::move(responseSink));
+yarpl::Reference<yarpl::flowable::Flowable<reactivesocket::Payload>>
+RSocketRequester::requestChannel(
+    yarpl::Reference<yarpl::flowable::Flowable<reactivesocket::Payload>>
+        payloads) {
+  auto& eb = eventBase_;
+  auto srs = reactiveSocket_;
+
+    LOG(INFO) << "requestChannel executing ";
+
+  return yarpl::flowable::Flowables::fromPublisher<Payload>([
+    &eb,
```

Comment:
Capturing one object field by reference is effectively the same as capturing the entire `this` pointer as you're depending on the entire object not being destroyed.  You're not really capturing the other objects fields by accident, whether they are or aren't in scope is the same question as "is eventBase_ in scope".

The issue I take with doing it this way is that it's harder to see the dependence on the object, as it requires jumping through `[..., &eb, ...]` and `auto& eb = eventBase_;`.